// baseui/tabs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Tabs", {
  enumerable: true,
  get: function get() {
    return _tabs.default;
  }
});
Object.defineProperty(exports, "StatefulTabs", {
  enumerable: true,
  get: function get() {
    return _statefulTabs.default;
  }
});
Object.defineProperty(exports, "Tab", {
  enumerable: true,
  get: function get() {
    return _tab.default;
  }
});
Object.defineProperty(exports, "ORIENTATION", {
  enumerable: true,
  get: function get() {
    return _constants.ORIENTATION;
  }
});
Object.defineProperty(exports, "STATE_CHANGE_TYPE", {
  enumerable: true,
  get: function get() {
    return _constants.STATE_CHANGE_TYPE;
  }
});
Object.defineProperty(exports, "StyledRoot", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Root;
  }
});
Object.defineProperty(exports, "StyledTab", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Tab;
  }
});
Object.defineProperty(exports, "StyledTabBar", {
  enumerable: true,
  get: function get() {
    return _styledComponents.TabBar;
  }
});
Object.defineProperty(exports, "StyledTabContent", {
  enumerable: true,
  get: function get() {
    return _styledComponents.TabContent;
  }
});

var _tabs = _interopRequireDefault(require("./tabs.js"));

var _statefulTabs = _interopRequireDefault(require("./stateful-tabs.js"));

var _tab = _interopRequireDefault(require("./tab.js"));

var _constants = require("./constants.js");

var _styledComponents = require("./styled-components.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
