// baseui/spinner/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "SpinnerDeterminate", {
  enumerable: true,
  get: function get() {
    return _index.ProgressBarRounded;
  }
});
Object.defineProperty(exports, "StyledSpinnerDeterminateRoot", {
  enumerable: true,
  get: function get() {
    return _index.StyledProgressBarRoundedRoot;
  }
});
Object.defineProperty(exports, "StyledSpinnerDeterminateSvg", {
  enumerable: true,
  get: function get() {
    return _index.StyledProgressBarRoundedSvg;
  }
});
Object.defineProperty(exports, "StyledSpinnerDeterminateText", {
  enumerable: true,
  get: function get() {
    return _index.StyledProgressBarRoundedText;
  }
});
Object.defineProperty(exports, "StyledSpinnerDeterminateTrackBackground", {
  enumerable: true,
  get: function get() {
    return _index.StyledProgressBarRoundedTrackBackground;
  }
});
Object.defineProperty(exports, "StyledSpinnerDeterminateTrackForeground", {
  enumerable: true,
  get: function get() {
    return _index.StyledProgressBarRoundedTrackForeground;
  }
});
Object.defineProperty(exports, "Spinner", {
  enumerable: true,
  get: function get() {
    return _spinner.default;
  }
});
Object.defineProperty(exports, "StyledSvg", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Svg;
  }
});
Object.defineProperty(exports, "StyledTrackPath", {
  enumerable: true,
  get: function get() {
    return _styledComponents.StyledTrackPath;
  }
});
Object.defineProperty(exports, "StyledActivePath", {
  enumerable: true,
  get: function get() {
    return _styledComponents.StyledActivePath;
  }
});
Object.defineProperty(exports, "StyledSpinnerNext", {
  enumerable: true,
  get: function get() {
    return _styledComponents.StyledSpinnerNext;
  }
});
Object.defineProperty(exports, "SIZE", {
  enumerable: true,
  get: function get() {
    return _constants.SIZE;
  }
});

var _index = require("../progress-bar/index.js");

var _spinner = _interopRequireDefault(require("./spinner.js"));

var _styledComponents = require("./styled-components.js");

var _constants = require("./constants.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
