// baseui/modal/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Modal", {
  enumerable: true,
  get: function get() {
    return _modal.default;
  }
});
Object.defineProperty(exports, "ModalButton", {
  enumerable: true,
  get: function get() {
    return _modalButton.default;
  }
});
Object.defineProperty(exports, "FocusOnce", {
  enumerable: true,
  get: function get() {
    return _focusOnce.default;
  }
});
Object.defineProperty(exports, "SIZE", {
  enumerable: true,
  get: function get() {
    return _constants.SIZE;
  }
});
Object.defineProperty(exports, "ROLE", {
  enumerable: true,
  get: function get() {
    return _constants.ROLE;
  }
});
Object.defineProperty(exports, "CLOSE_SOURCE", {
  enumerable: true,
  get: function get() {
    return _constants.CLOSE_SOURCE;
  }
});
Object.defineProperty(exports, "StyledRoot", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Root;
  }
});
Object.defineProperty(exports, "StyledBackdrop", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Backdrop;
  }
});
Object.defineProperty(exports, "StyledDialog", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Dialog;
  }
});
Object.defineProperty(exports, "StyledDialogContainer", {
  enumerable: true,
  get: function get() {
    return _styledComponents.DialogContainer;
  }
});
Object.defineProperty(exports, "StyledClose", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Close;
  }
});
Object.defineProperty(exports, "ModalHeader", {
  enumerable: true,
  get: function get() {
    return _styledComponents.ModalHeader;
  }
});
Object.defineProperty(exports, "ModalBody", {
  enumerable: true,
  get: function get() {
    return _styledComponents.ModalBody;
  }
});
Object.defineProperty(exports, "ModalFooter", {
  enumerable: true,
  get: function get() {
    return _styledComponents.ModalFooter;
  }
});

var _modal = _interopRequireDefault(require("./modal.js"));

var _modalButton = _interopRequireDefault(require("./modal-button.js"));

var _focusOnce = _interopRequireDefault(require("./focus-once.js"));

var _constants = require("./constants.js");

var _styledComponents = require("./styled-components.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
