// baseui/tag/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Tag", {
  enumerable: true,
  get: function get() {
    return _tag.default;
  }
});
Object.defineProperty(exports, "StyledRoot", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Root;
  }
});
Object.defineProperty(exports, "StyledAction", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Action;
  }
});
Object.defineProperty(exports, "StyledActionIcon", {
  enumerable: true,
  get: function get() {
    return _styledComponents.ActionIcon;
  }
});
Object.defineProperty(exports, "StyledText", {
  enumerable: true,
  get: function get() {
    return _styledComponents.Text;
  }
});
Object.defineProperty(exports, "KIND", {
  enumerable: true,
  get: function get() {
    return _constants.KIND;
  }
});
Object.defineProperty(exports, "VARIANT", {
  enumerable: true,
  get: function get() {
    return _constants.VARIANT;
  }
});
Object.defineProperty(exports, "SIZE", {
  enumerable: true,
  get: function get() {
    return _constants.SIZE;
  }
});

var _tag = _interopRequireDefault(require("./tag.js"));

var _styledComponents = require("./styled-components.js");

var _constants = require("./constants.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
